src/higlass/widget.js

```diff
@@ -1,4 +1,129 @@
 import * as hglib from "https://esm.sh/higlass@1.13?deps=react@17,react-dom@17,pixi.js@6";
+import { v4 } from "https://esm.sh/@lukeed/uuid@2.0.1";
+
+/** @import { HGC, PluginDataFetcherConstructor } from "./types.ts" */
+
+// Make sure plugins are registered and enabled
+window.higlassDataFetchersByType = window.higlassDataFetchersByType ||
+  {};
+
+/**
+ * Create a unique identifier.
+ * @returns {string}
+ */
+function uid() {
+  return v4().split("-")[0];
+}
+
+/**
+ * Make an assertion.
+ *
+ * @param {unknown} expression - The expression to test.
+ * @param {string=} msg - The optional message to display if the assertion fails.
+ * @returns {asserts expression}
+ * @throws an {@link Error} if `expression` is not truthy.
+ */
+function assert(expression, msg = "") {
+  if (!expression) throw new Error(msg);
+}
+
+/**
+ * @template T
+ * @param {import("npm:@anywidget/types").AnyModel} model
+ * @param {unknown} payload
+ * @param {{ timeout?: number }} [options]
+ * @returns {Promise<{ data: T, buffers: DataView[] }>}
+ */
+function send(model, payload, { timeout = 3000 } = {}) {
```

Comment:
I'm not sure I fully grok what this function does.

Reply:
There are two ways to send data between JS and Python: traitlets & [custom messages](https://anywidget.dev/en/jupyter-widgets-the-good-parts/#connecting-javascript-with-python).

Custom messages are normally "fire and forget", so we need some kind of mechanism to fire and "await" a response back from Python. I've implemented an experimental API for this in anywidget, trying to make this pattern easier... but the implementation is pretty straigtht forward and included it here to avoid relying on an experimental API.